lib/ensure-default-attributes.coffee

```diff
@@ -17,25 +17,26 @@ class EnsureDefaultAttributes
     updatedVariants = _.map(product.variants,
       (variant) =>
         if productFromServer
-          serverVariant = productFromServer.variants.filter((v) => v.sku == variant.sku)[0]
+          serverVariant = productFromServer.variants.filter((v) -> v.sku == variant.sku)[0]
         @_ensureInVariant(variant, serverVariant)
     )
     updatedProduct.variants = updatedVariants
     Promise.resolve(updatedProduct)
 
   _ensureInVariant: (variant, serverVariant) =>
+    defaultAttributes = _.deepClone(@defaultAttributes)
```

Comment:
let's add a comment why we are doing this in the code...

Took me a while to figure it out